rest-api/api/participant_counts_over_time_api.py

```diff
@@ -0,0 +1,102 @@
+import datetime
+# import json
+# import logging
+
+from flask.ext.restful import Resource
+from flask import request
+from werkzeug.exceptions import BadRequest
+
+from api_util import HEALTHPRO
+from api_util import get_awardee_id_from_name
+from app_util import auth_required
+from dao.hpo_dao import HPODao
+from participant_enums import EnrollmentStatus
+
+DATE_FORMAT = '%Y-%m-%d'
+DAYS_LIMIT = 100  # provisional, per design doc
+
+
+class ParticipantCountsOverTimeApi(Resource):
+
+  @auth_required(HEALTHPRO)
+  def get(self):
+    self.hpo_dao = HPODao()
+
+    # TODO: After enrollment status is filterable,
+    # wire in 'organization', 'site', and 'enrollmentStatus'.
+    enrollment_status = request.args.get('enrollmentStatus')
+    awardee = request.args.get('awardee')
+    stratification = request.args.get('stratification')
+    start_date = request.args.get('startDate')
+    end_date = request.args.get('endDate')
+
+    params = {
+      'enrollment_statuses': enrollment_status,
+      'awardees': awardee,
+      'stratification': stratification,
+      'start_date': start_date,
+      'end_date': end_date
+    }
+
+    # Most parameters accepted by this API can have multiple, comma-delimited
+    # values.  Arrange them into lists.
+    for param in params:
+      value = params[param]
+      if param in ['start_date', 'end_date', 'stratification']:
```

Comment:
I think doing this will always try to encode these three values. If it's not provided do you get an error?

Reply:
I thought I encountered an issue that encoding fixed, but I can't reproduce it, so I've removed the unneeded encoding.